worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -153,6 +154,15 @@ function routeTask(rootDocument) {
 
                 rootDocument.getField('CAF_WORKFLOW_ACTION').add(action.name);
                 applyActionDetails(rootDocument, actionDetails, terminateOnFailure);
+                if(action.applyMessagePrioritization) {
+                    if(!MessageRouterSingleton) {
+                        var mrs = 
+                            Java.type("com.github.workerframework.workermessageprioritization.rerouting.MessageRouterSingleton");
+                        mrs.init();
+                        MessageRouterSingleton = mrs;
+                    }
+                    MessageRouterSingleton.route(rootDocument);
+                }
```

Comment:
What is the intention here?  Javascript variables will be task-scope so this `MessageRouterSingleton` will be start at `undefined` for each task flowing through the worker.  The `if(!MessageRouterSingleton)` check will not prevent the `mrs.init();` call being made from two threads if that is the intention.

Reply:
Ok, if the variable is scoped to the task this is pointless.
Multiple calls to init() are safe this was an attempt to avoid calling Java.type() multiple times, which given what you said isn't going to be possible.

Reply:
It will reuse the Javascript engine across the tasks so I wouldn't anticipate the `Java.type()` call being an issue.

Reply:
I was going to ask for CAF_WMP_ENABLED ENV to be checked in this js as I was concerned about a worker that had't been updated being asked to load the MessageRouterSingleton type by the workflow script. 

@dermot-hardy 
Are we comfortable that with the latest worker framework version that all workers, that receive the new workflow script, will have a MessageRouterSingleton on their classpath and we don't need to check CAF_WMP_ENABLED in workflow-control.js?

Reply:
> Are we comfortable that... we don't need to check `CAF_WMP_ENABLED` in `workflow-control.js`?

No we should check it.

Reply:
I've updated this now in this commit https://github.com/CAFDataProcessing/worker-workflow/pull/107/commits/188cf95ba24633cae80944042c784012ec34d6a8